holidays/ical.py

```diff
@@ -0,0 +1,140 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+import uuid
+from datetime import datetime, timezone
+
+
+class ICalExporter:
+    def __init__(self, holidays_object, return_bytes=False, ical_timestamp=None):
+        """
+        Initialize iCalendar exporter
+
+        Args:
+        - holidays_object: Holidays object containing holiday data.
+        - return_bytes(bool): If True, return bytes instead of string.
+        - ical_timestamp: Optional override for the iCal "DTSTAMP" parameter.
+            If not provided, current datetime (UTC timezone) will be used instead.
+        """
+        self.holidays = holidays_object
+        self.return_bytes = return_bytes
+        if ical_timestamp is None:
+            self.ical_timestamp = datetime.now().astimezone(timezone.utc).strftime("%Y%m%dT%H%M%S")
+        else:
+            self.ical_timestamp = ical_timestamp
+
+        try:
+            with open("holidays/version.py") as file:
+                for line in file:
+                    if line.startswith("__version__"):
+                        self.holidays_version = line.split("=")[1].strip().strip('"')
+                        break
+                else:
+                    raise ValueError("Version not found in holidays/version.py")
+        except FileNotFoundError:
+            raise FileNotFoundError("The file 'holidays/version.py' was not found.")
+
+        self.language = self._validate_language(getattr(self.holidays, "language", "EN"))
+
+    def _validate_language(self, language):
+        """
+        Validates the language code. Ensures it's a two-letter ISO 639-1 code
+        as iCal only supports that at the moment.
+
+        Args:
+        - language: The language code to validate.
+
+        Returns:
+        - Valid two-letter language code in uppercase.
+        """
+        language = (language or "EN").upper().split("_")[0]
```

Comment:
```suggestion
        language = language.upper().split("_")[0]
```
language is not empty on method calling.

Reply:
I'm reverting this back for now since it affects non-`country/financial_holidays` iCal export i.e. the ones used in testcases:
```python
        # Keep this for when we support multiple-day length iCal export.
        test_single_holiday_continuous = {
            date(2024, 4, 13): "Songkran",
            date(2024, 4, 14): "Songkran",
            date(2024, 4, 15): "Songkran",
        }
        exporter = ICalExporter(test_single_holiday_continuous)
```

Reply:
I ended up converting all test case dicts to `MockHolidays` object instead so this is now solved.